lib/bolt_server/plan_app.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'bolt'
+require 'bolt/error'
+require 'bolt/executor'
+require 'bolt/inventory'
+require 'bolt/pal'
+require 'concurrent'
+require 'json'
+require 'json-schema'
+
+module BoltServer
+  class PlanApp < Sinatra::Base
+    # This disables Sinatra's error page generation
+    set :show_exceptions, false
+
+    def initialize(modulepath)
+      @schema = JSON.parse(File.read(File.join(__dir__, 'schemas', 'run_plan.json')))
+      @worker = Concurrent::SingleThreadExecutor.new
+
+      # Create a basic executor, leave concurrency up to Orchestrator.
+      @executor = Bolt::Executor.new(0)
+      # Use an empty inventory until we figure out where this data comes from.
+      @inventory = Bolt::Inventory.new(nil)
+      # TODO: what should max compiles be set to for apply?
+      @pal = Bolt::PAL.new(modulepath, nil)
+
+      @last_result = Concurrent::Promise.new(executor: @worker) { nil }
```

Comment:
I'm not sure I understand what this promise is for, or if we still need it? Isn't this a part of the `/plan/run` handler?

Reply:
This was to make sure I didn't have to check whether @last_result was set before accessing it.